guides/creating-a-custom-form-field-control.md

```diff
@@ -0,0 +1,337 @@
+type we used for the type parameter when we implemented `MatFormFieldControl`. Since our component 
+already has a value property we don't need to do anything for this one.
+
+#### `stateChanges`
+Because the `<mat-form-field>` used the `OnPush` change detection strategy, we need to let it know
+when something happens in the form field control that may require the form field to run change
+detection. We do this via the `stateChanges` property. So far the only thing the form field needs to
+know about is when the value changes. We'll need to emit on the stateChanges stream when that
+happens, and as we continue flushing out these properties we'll likely find more places we need to
+emit. We should also make sure to complete `stateChanges` when our component is destroyed
+
+```ts
+stateChanges = new Subject<void>();
+
+set value(tel: MyTel | null) {
+  ...
+  stateChanges.next();
+}
+
+ngOnDestory() {
+  stateChanges.destroy();
+}
+```
+
+#### `id`
+This property should return the ID of an element in the component's template that we want the
+`<mat-form-field>` to associate all of its labels and hints with. In this case we'll use the host
+element and just generate a unique ID for it.
+
+```ts
+static nextId = 0;
+
+@HostBinding() id = `my-tel-input-${MyTelInput.nextId++}`;
+```
+
+#### `placeholder`
+This property allows us to tell the `<mat-form-field>` what to use as a placeholder. In this
+example, we'll do the same thing as `matInput` and `<mat-select>` and allow the user to specify it
+via an `@Input()`. Since the value of the placeholder may change over time, we need to make sure to
+trigger change detection in the parent form field by emitting on the `stateChanges` stream when the
+placeholder changes.
+
+```ts
+@Input()
+get placeholder() {
+  return this._placeholder;
+}
+set placeholder(plh) {
+  this._placeholder = plh;
+  this.stateChanges.next();
+}
+private _placeholder: string;
+```
+
+#### `ngControl`
+This property allows the form field control to specify the `@angular/forms` control that is bound to
+this component. Since we haven't set up our component to act as a `ControlValueAccessor` we'll just
+set this to `null` in our component. In any real component you would probably want to implement
+`ControlValueAccessor` so that your component can work with `formControl` and `ngModel`.
+
+```ts
+ngControl = null;
+```
+
+If you did implement `ControlValueAccessor`, you could simply inject the `NgControl` and make it
+publicly available. (For additional information about `ControlValueAccessor` see the
+[API docs](https://angular.io/api/forms/ControlValueAccessor).)
+
+```ts
+constructor(..., @Optional() @Self() public ngControl: NgControl) { ... }
+```
+
+#### `focused`
+This property indicates whether or not the form field control should be considered to be in a
+focused state. When it is in a focused state the form field is displayed with a solid color
```

Comment:
When it is in a focused state the form field -> When it is in a focused state, the form field